sdk-python/copilotkit/langgraph_agent.py

```
@@ -366,116 +366,153 @@ async def _stream_events( # pylint: disable=too-many-locals
-                    thread_id=thread_id,
-                    run_id=run_id,
-                    node_name=node_name,
-                    state=manually_emitted_state,
-                    running=True,
-                    active=True
-                ) + "\n"
-                continue
 
+                # we only want to update the node name under certain conditions
+                # since we don't need any internal node names to be sent to the frontend
+                if current_node_name in self.graph.nodes.keys():
```

Ruff (0.11.9)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)